src/commcare_cloud/commands/ansible/run_module.py

```diff
@@ -138,6 +138,7 @@ def run_ansible_module(ansible_context, inventory_group, module, module_args,
             '-e', '@{}'.format(environment.paths.generated_yml),
         )
         cmd_parts += environment.secrets_backend.get_extra_ansible_args()
+    if needs_secrets or environment.secrets_backend.has_extra_env_vars():
```

Comment:
I think I see what you're doing—insert a special `ANSIBLE_SSH_EXTRA_ARGS` value if this is an AWS environment—but the secrets_backend seems like kind of an awkward place to graft that in. The secrets backend's job is to mediate fetching/storing environment secrets, not connection settings. Anything I'm missing about how this is logically tied to e.g. the choice to use AWS Secrets Manager vs ansible-vault for storing secrest?

If there's not a nuance here I'm missing, I might suggest putting these methods directly on `environment`—even though that's already somewhat cluttered, I think these methods go better there. I'm imagining the env vars would be additive—so first you'd add the env vars related to the secrets backend, and then separately you'd add the env vars related to the connection.

Does that make sense?

Reply:
Yes, that makes sense, and I like the idea.

Reply:
https://github.com/dimagi/commcare-cloud/pull/5602/commits/8fc4a9ab31f663f24b3f6882085d96a6d085ce72 - best viewed with [this diff](https://github.com/dimagi/commcare-cloud/compare/09fb356b9761bfd3b0ee6442b3899e99c01cfd2f..8fc4a9ab31f663f24b3f6882085d96a6d085ce72), ignoring changes to `hosting_docs/source/reference/1-commcare-cloud/commands.md` from https://github.com/dimagi/commcare-cloud/pull/5602/commits/0e3047f84e5aaafdc0560a7bddefb9478bf9d2d2